src/dev/flang/ast/AstErrors.java

```diff
@@ -1038,6 +1038,39 @@ static String solutionDeclareReturnTypeIfResult(String name, int argCount)
     return solution;
   }
 
+  /**
+   * Suggest to a user that they might have called a feature with a wrong number of arguments,
+   * in the case that the called feature could not be found but there is a feature with the same
+   * name but a differing number of arguments.
+   */
+  static String solutionWrongArgumentNumber(FeatureName name,
+                                            List<FeatureAndOuter> l)
+  {
+    var solution = "";
+    var found = new List<FeatureName>();
+
+    for (var fo : l)
+      {
+        var f = fo._feature;
+        var fn = f.featureName();
+        if (f.isChoice() && !f.isBaseChoice())
+          {
+            /* ignore call to choice type */
+          }
+        else if (fn.equalsBaseName(name))
+          {
+            found.add(fn);
+          }
+      }
+
+    if (!found.isEmpty())
+      {
+        solution = "Did you perhaps mean to use " + (found.size() > 1 ? "one of these features" : "this feature") + ": " + found + "?";
```

Comment:
Please reformulate this and start with `To solve this, `, this is used in all other error messages when proposing a workaround: something like `To solve this, you might change the actual number of arguments to match ...